lib/galaxy/model/database_heartbeat.py

```diff
@@ -79,7 +80,7 @@ def worker_process(self):
     def update_watcher_designation(self):
         worker_process = self.worker_process
         if not worker_process:
-            worker_process = WorkerProcess(server_name=self.server_name, hostname=self.hostname)
+            worker_process = WorkerProcess(server_name=self.server_name, hostname=self.hostname, pid=os.getpid())
```

Comment:
We also need some logic that updates worker pids, this will only do it for new workers (meaning a new server_name/hostname combination). Maybe add a pid attribute to the DatabaseHeartbeat. Since this needs to happen after forking I'd set this to None in `__init__` and only set the pid in `start `. Then you can drop the new argument to `WorkerProcess` on this line and just do `worker_process.pid = self.pid`. I think that should work.